### PR TITLE
Handle VLRs as pipeline arguments in writers.las

### LINKED ISSUE #2937: writers.las can't accept a list of VLRs from a pipeline
Because m_userVLRs is a JSON blob, the pipeline reader can't set a list of VLRs in the pipeline.

### PR DESCRIPTION
**Pull request by abellgithub**
Close #2937

### CODE REVIEW

File: io/LasVLR.cpp [hunk 1]
```diff
@@ -57,6 +62,115 @@ ILeStream& operator>>(ILeStream& in, LasVLR& v)
...
+        throw pdal_error("LAS VLR must contain 'data' member.");
+    if (userId.empty())
+        throw pdal_error("LAS VLR must contain 'user_id' member.");
+    if (std::isnan(recordId))
+        recordId = 1;
+
+    v.m_userId = userId;
+    v.m_recordId = recordId;
+    v.m_description = description;
+    v.m_data = Utils::base64_decode(b64data);
```
Comment (hobu) [inline io/LasVLR.cpp]: Do we need to protect against too large of a size?

Comment (abellgithub) [reply io/LasVLR.cpp]: The data?

Comment (hobu) [reply io/LasVLR.cpp]: Sorry, yeah. The data's size needs to be under a VLR.

Comment (abellgithub) [reply io/LasVLR.cpp]: It can be as large as any 64 bit integer value, right?  So, no, if that's what you mean.

Comment (hobu) [reply io/LasVLR.cpp]: EVLRs can be as large as any 64 bit integer value. VLRs can only be as large as a 16 bit value.

Comment (abellgithub) [reply io/LasVLR.cpp]: True.

We treat everything as an EVLR if it's over the VLR limit.  If we can't do that (pre 1.4), then we throw an error at the time we try to add the VLR to the list of VLRs.  But I think things are fine here.

Comment (hobu) [reply io/LasVLR.cpp]: Ok. Anyone pushing 4gb of base64 through pipeline JSON deserves their fate anyway.

Comment (abellgithub) [reply io/LasVLR.cpp]: https://github.com/PDAL/PDAL/blob/227855a35532b7e978245a6e1a1072aa00bfd36a/io/LasWriter.cpp#L573-L593

**Pull request merged**